sbin/spark-daemon.sh

```diff
@@ -164,9 +165,10 @@ case $option in
   (stop)
 
     if [ -f $pid ]; then
-      if kill -0 `cat $pid` > /dev/null 2>&1; then
+      TARGET_ID=`cat $pid`
+      if [[ `ps -p $TARGET_ID -o args=` =~ $command ]]; then
         echo stopping $command
-        kill `cat $pid`
+        kill $TARGET_ID
       else
         echo no $command to stop
```

Comment:
I know this was not part of your changes, but we should also quote this whole string here, too.

```
echo "no $command to stop"
```